packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1142,88 +1142,136 @@ bool hasPlugins(FlutterProject project) {
+Map<String, List<Plugin>> _resolvePluginImplementations(
+  List<Plugin> plugins, {
+  required bool selectDartPluginsOnly,
+}) {
+  final Map<String, List<Plugin>> pluginsByPlatform = <String, List<Plugin>>{
+    AndroidPlugin.kConfigKey: <Plugin>[],
+    IOSPlugin.kConfigKey: <Plugin>[],
+    LinuxPlugin.kConfigKey: <Plugin>[],
+    MacOSPlugin.kConfigKey: <Plugin>[],
+    WindowsPlugin.kConfigKey: <Plugin>[],
+    WebPlugin.kConfigKey: <Plugin>[],
+  };
 
-      if (defaultImplPluginName != null) {
-        // Each plugin can only have one default implementation for this [platformKey].
-        defaultImplementations[plugin.name] = defaultImplPluginName;
-      }
-      if (implementsPluginName != null) {
-        pluginImplCandidates.putIfAbsent(implementsPluginName, () => <Plugin>[]);
-        pluginImplCandidates[implementsPluginName]!.add(plugin);
-      }
-    }
+  bool hasPluginPubspecError = false;
+  bool hasResolutionError = false;
 
-    final Map<String, Plugin> pluginResolution = <String, Plugin>{};
+  for (final String platformKey in pluginsByPlatform.keys) {
+    final (
+      List<Plugin> platformPluginResolutions,
+      bool hasPlatformPluginPubspecError,
+      bool hasPlatformResolutionError
+    ) = _resolvePluginImplementationsByPlatform(
+      plugins,
+      platformKey,
+      selectDartPluginsOnly: selectDartPluginsOnly,
+    );
 
-    // Now resolve all the possible resolutions to a single option for each
-    // plugin, or throw if that's not possible.
-    for (final MapEntry<String, List<Plugin>> implCandidatesEntry in pluginImplCandidates.entries) {
-      final (Plugin? resolution, String? error) = _resolveImplementationOfPlugin(
-        platformKey: platformKey,
-        pluginName: implCandidatesEntry.key,
-        candidates: implCandidatesEntry.value,
-        defaultPackageName: defaultImplementations[implCandidatesEntry.key],
-      );
-      if (error != null) {
-        globals.printError(error);
-        hasResolutionError = true;
-      } else if (resolution != null) {
-        pluginResolution[implCandidatesEntry.key] = resolution;
-      }
+    if (hasPlatformPluginPubspecError) {
+      hasPluginPubspecError = true;
+    } else if (hasPlatformResolutionError) {
+      hasResolutionError = true;
+    } else {
+      pluginsByPlatform[platformKey] = platformPluginResolutions;
     }
-
-    pluginResolutions.addAll(
-      pluginResolution.values.map((Plugin plugin) {
-        return PluginInterfaceResolution(plugin: plugin, platform: platformKey);
-      }),
-    );
   }
   if (hasPluginPubspecError) {
     throwToolExit('Please resolve the plugin pubspec errors');
   }
   if (hasResolutionError) {
     throwToolExit('Please resolve the plugin implementation selection errors');
   }
-  return pluginResolutions;
+  return pluginsByPlatform;
+}
+
+/// Resolves the plugins for the given [platformKey] (dart-only or native
```

Comment:
Dart-only

Reply:
Done in f2a85fdfc2dc0e690bf03256bcda960e20067d86